src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,138 @@
+            }
+
+        } else if (userInput.startsWith("delete")) {
+            String[] parts = userInput.split(" ");
+            if (parts.length == 2) {
+                try {
+                    int taskToDelete = Integer.parseInt(parts[1]) - 1;
+                    if (taskToDelete > taskList.getBotMemory().size() - 1 || taskToDelete < 0) {
+                        throw new MollyException("Please select a valid task number.");
+                    } else {
+                        taskList.removeTask(taskToDelete);
+                        storage.saveTasks(taskList);
+                    }
+                } catch (NumberFormatException e) {
+                    Ui.printLine();
+                    throw new MollyException("Invalid command. Please enter a valid task number.");
+                }
+            } else {
+                Ui.printLine();
+                throw new MollyException("Invalid  delete command. Please use 'delete (task number)'.");
+            }
+
+        } else if (userInput.startsWith("find")) {
+            String[] queryParts = userInput.split(" ");
+            if (queryParts.length == 2) {
+                String searchTerm = queryParts[1];
+                taskList.findTerm(searchTerm);
+            } else {
+                Ui.printLine();
+                throw new MollyException("Invalid find command. Please use 'find (search term)'.");
+            }
+
+        } else if (!userInput.toLowerCase().equals("list")) {
+            if (userInput.startsWith("todo")) {
+                if (userInput.equals("todo")) {
+                    Ui.printLine();
+                    throw new MollyException("Yikes! Sorry, the todo description cannot be empty.");
+                } else {
+                    Task newToDo = new Task(userInput.substring(5));
+                    taskList.addTask(newToDo);
+                    storage.saveTasks(taskList);
+                }
+            } else if (userInput.startsWith("deadline")) {
+                if (userInput.equals("deadline")) {
+                    Ui.printLine();
+                    throw new MollyException("Yikes! Sorry, the description of a deadline cannot be empty.");
+                } else {
+                    String[] deadlineDetails = userInput.substring(9).split( " /by ");
+                    if (deadlineDetails.length == 2) {
+                        String description = deadlineDetails[0].trim();
+                        String by = deadlineDetails[1].trim();
+                        Deadline newDeadline = new Deadline(description, by);
+                        taskList.addTask(newDeadline);
+                        storage.saveTasks(taskList);
+                    } else {
+                        Ui.printLine();
+                        throw new MollyException("Sorry, invalid format for deadline. Please follow the format: deadline (description) /by (end date). The end date can be in the format DD-MM-YYYY HHmm (24 hour format).");
```

Comment:
The coding standard recommends keeping the line length shorter than 110 characters, and has a hard limit of 120 characters. For long messages like this, it might be better to do line wrapping to improve readability.
```suggestion
                        throw new MollyException("Sorry, invalid format for deadline. "
                                + "Please follow the format: deadline (description) /by (end date). "
                                + "The end date can be in the format DD-MM-YYYY HHmm (24 hour format).");
```